src/logs/sentry.rs

```diff
@@ -94,119 +104,183 @@ impl IntoIterator for LazySentryLogs {
     }
 }
 
+pub trait Stat {
+    fn check_raw_log(&mut self, log: &RawLog);
+    fn report(&self);
+    fn process_raw_logs<'l>(&mut self, logs: impl Iterator<Item = &'l RawLog>) {
+        for log in logs {
+            self.check_raw_log(log);
+        }
+    }
+}
+
+pub struct SuccessfulScan {
+    pub total: usize,
+}
+
+pub struct MalformedQr {
+    pub total: usize,
+}
+
+pub struct RegexMatch {
+    key: String,
+    re: Regex,
+    total_checked: usize,
+    pub matches: usize,
+}
+
+pub enum SentryLogsStatChecker {
+    SuccessfulScans(SuccessfulScan),
+    MalformedQr(MalformedQr),
+    RegexMatch(RegexMatch),
+}
+
+pub struct SentryLogsStatsExecutor(Vec<SentryLogsStatChecker>);
+
 #[derive(Debug, Clone)]
 pub struct SentryFragmentLog {
     pub public_key: String,
     pub chain_proposal_index: u8,
-    pub proposal_index: u8,
+    pub proposal_index: u32,
     pub voteplan_id: String,
     pub choice: u8,
     pub spending_counter: u64,
     pub fragment_id: String,
 }
 
-#[derive(Debug, Eq, PartialEq)]
-pub struct LogCmpFields {
-    pub public_key: String,
-    pub chain_proposal_index: u8,
-    pub voteplan_id: String,
-    pub choice: u8,
-    pub fragment_id: String,
+fn raw_log_message_starts_with(log: &RawLog, pattern: &str) -> bool {
+    log.get("message")
+        .and_then(|message| message.as_str().map(|s| s.starts_with(pattern)))
+        .unwrap_or(false)
 }
 
-impl From<SentryFragmentLog> for LogCmpFields {
-    fn from(log: SentryFragmentLog) -> Self {
-        Self {
-            public_key: log.public_key,
-            chain_proposal_index: log.chain_proposal_index,
-            voteplan_id: log.voteplan_id,
-            choice: log.choice,
-            fragment_id: log.fragment_id,
+impl Stat for SuccessfulScan {
+    fn check_raw_log(&mut self, log: &RawLog) {
+        if raw_log_message_starts_with(log, REGISTERED_MESSAGE) {
+            self.total += 1;
         }
     }
+
+    fn report(&self) {
```

Comment:
Maybe pass a buffer or similar like `std::fmt::Formatter` to be a little more flexible if we don't want to pollute stdout?

Reply:
Aah, yeah, that is indeed better though. Will do.